pkg/connector/user.go

```diff
@@ -138,23 +142,48 @@ func (u *userResourceType) List(ctx context.Context, _ *v2.ResourceId, pt *pagin
 		return nil, "", nil, fmt.Errorf("onelogin-connector: failed to list users: %w", err)
 	}
 
-	nextPage, err := bag.NextToken(nextCursor)
-	if err != nil {
-		return nil, "", nil, err
-	}
-
+	displayNameCache := make(map[int]string)
 	var rv []*v2.Resource
-	for _, user := range users {
-		userCopy := user
-		ur, err := userResource(ctx, &userCopy)
 
+	for i := range users {
+		user := &users[i]
+
+		fullUser, err := u.client.GetUserByID(ctx, user.Id)
 		if err != nil {
-			return nil, "", nil, err
+			log.Printf("Error obtaining user %d: %v", user.Id, err)
```

Comment:
AFAIK, you are not supposed to use `log` from the standard library, instead you should use zap. If you passed a `context.Context` to this method/function, you can extract the logger with `l := ctxzap.Extract(ctx)` somewhere up at the beginning of your function, _then_ you would log that line as `l.Info("error obtaining user", zap.Int("user", user.Id), zap.Error(err))`